buildpacks/maven/bin/build

```diff
@@ -185,6 +185,11 @@ log::cnb::debug "Internal Maven options: ${internal_maven_options[*]}"
 export MAVEN_OPTS="${MAVEN_JAVA_OPTS:-"-Xmx1024m"}"
 log::cnb::debug "Setting MAVEN_OPTS=${MAVEN_OPTS}"
 
+# Ensure customer vendored maven is executable
+if [ -f "${maven_executable}" ]; then
+	chmod +x "${maven_executable}"
+fi
+
```

Comment:
What do you think about moving this inside the block where the customer vendored wrapper is detected/selected?
https://github.com/heroku/buildpacks-jvm/blob/b6a880864860f8beab5dd1bc9e1c81899e2550ba/buildpacks/maven/bin/build#L116

With the current distance between that block and this, it would be easy for another `maven_executable` usage to sneak in that's run before the permissions are updated. Also I'm presuming the chmod only ever needs to be run for the customer vendored case, whereas here it runs for all builds instead.

Reply:
Good idea. I moved it.